container-engine-lib/lib/backend_impls/kubernetes/kubernetes_manager/kubernetes_manager.go

```diff
@@ -1482,6 +1484,90 @@ func (manager *KubernetesManager) RunExecCommand(
 	return successExecCommandExitCode, nil
 }
 
+func (manager *KubernetesManager) RunExecCommandWithStreamedOutput(
+	ctx context.Context,
+	namespaceName string,
+	podName string,
+	containerName string,
+	command []string,
+) (chan string, chan *exec_result.ExecResult, error) {
+	execOptions := &apiv1.PodExecOptions{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "",
+			APIVersion: "",
+		},
+		Stdin:     false,
+		Stdout:    true,
+		Stderr:    true,
+		TTY:       true,
+		Container: containerName,
+		Command:   command,
+	}
+
+	//Create a RESTful command request.
+	request := manager.kubernetesClientSet.CoreV1().RESTClient().
+		Post().
+		Namespace(namespaceName).
+		Resource("pods").
+		Name(podName).
+		SubResource("exec").
+		VersionedParams(execOptions, scheme.ParameterCodec)
+	if request == nil {
+		return nil, nil, stacktrace.NewError("Failed to build a working RESTful request for the command '%s'.", execOptions.Command)
+	}
+
+	exec, err := remotecommand.NewSPDYExecutor(manager.kuberneteRestConfig, http.MethodPost, request.URL())
+	if err != nil {
+		return nil, nil, stacktrace.Propagate(err,
+			"Failed to build an executor for the command '%s' with the RESTful endpoint '%s'.",
+			execOptions.Command,
+			request.URL().String())
+	}
+
+	execOutputChan := make(chan string)
+	finalExecResultChan := make(chan *exec_result.ExecResult)
+	go func() {
+		defer func() {
+			close(execOutputChan)
+			close(finalExecResultChan)
+		}()
+
+		// Stream output from k8s to output channel
+		outputBuffer := &bytes.Buffer{}
+		streamWriter := stream_writer.NewStreamWriter(outputBuffer, execOutputChan)
```

Comment:
Coolios so here you don't need the outputBuffer anymore, right? You can implement a simple ChannelWriter, something like:
```
type ChannelWriter struct {
	channel chan string
}

func (writer *ChannelWriter) Write(input []byte) (int, error) {
	writer.channel <- string(input)
	return len(input), nil
}
```
which is a bit simpler than the StreamWriter (no lock)

Reply:
yes this is much better, will change